src/routes/safe/components/Transactions/TxList/ActionModal.tsx

```diff
@@ -32,19 +32,9 @@ export const ActionModal = (): ReactElement | null => {
       return <RejectTxModal isOpen onClose={onClose} gwTransaction={transaction} />
 
     case 'confirm':
-      return (
-        <ApproveTxModal
-          isOpen
-          onClose={onClose}
-          transaction={transaction as Overwrite<Transaction, { txDetails: ExpandedTxDetails }>}
-          txParameters={txParameters}
-        />
-      )
-
     case 'execute':
       return (
         <ApproveTxModal
-          isExecution
```

Comment:
How do you differentiate whether it is an execution or not?

Reply:
Inside the modal itself, relying on whether the tx can be executed at all and the checkbox.